src/anm/analytics.js

```diff
@@ -0,0 +1,82 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
+            } else {
+                var trackUrl = animatronUrl + '?data=' + encodeURIComponent(data);
+                sendViaGif(trackUrl);
+            }
+        } else {
+            setTimeout(event, timeout);
```

Comment:
First, why the timeout is ever different for `sendBeacon`? Just in case?

This call is performed when queue is empty, no matter was `sendBeacon` supported or not, why should timeout be different here?